src/ucp/wireup/wireup_cm.c

```diff
@@ -312,24 +311,23 @@ static unsigned ucp_cm_client_connect_progress(void *arg)
 
     status = uct_cm_client_ep_conn_notify(uct_cm_ep);
     if (status != UCS_OK) {
+        ucp_ep->flags &= ~UCP_EP_FLAG_LOCAL_CONNECTED;
```

Comment:
isn't set_ep_failed() doing this already?
seems removing this flag will prevent set_ep_failed from removing CM lane, why is this what we want?

Reply:
if the flag is set set_ep_failed() will call disconnect on failed EP, this is not what we want (UCT will return error)
(failed and disconnected lane is not the same)

Reply:
pls add code comment to explain